RIOT-OS/RIOT
# lwip: provide sock_async support

### Contribution description
This provides initial `sock_async` support for lwIP. ~~Currently only `sock_udp` and `sock_ip` are supported, `sock_tcp` needs some extra work, I don't have the time for right now.~~ (edit: was a bit tricky but not that hard after all ;-)) The change requires an [update to upstream](https://savannah.nongnu.org/patch/index.php?9891), which I provided in this PR as a patch.



### Testing procedure
I updated `tests/lwip` to use `sock_async_event` for the `udp server` and `ip server` [*edit: and the `tcp server`*] command, so running `make -C tests/lwip all test` on `native` should still work. 



### Issues/PRs references
None.
